validator/fw/string/is_url_with_http_or_https.go

```diff
@@ -0,0 +1,44 @@
+package string
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/helpers/validatordiag"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"golang.org/x/exp/slices"
+)
+
+// Ensure our implementation satisfies the validator.String interface.
+var _ validator.String = &urlHttpOrHttpsValidator{}
+
+type urlHttpOrHttpsValidator struct{}
+
+func (v urlHttpOrHttpsValidator) Description(_ context.Context) string {
```

Comment:
So, the idea in TF was to expand how validation was done and put an interface on it?